implementations/rust/ockam/ockam_command/src/project/util.rs

```diff
@@ -217,48 +217,62 @@ pub async fn check_project_readiness<'a>(
 }
 
 pub mod config {
-    use ockam::Context;
+    use ockam::{identity::PublicIdentity, Context};
+    use ockam_api::config::lookup::ProjectAuthority;
+    use ockam_vault::Vault;
 
     use super::*;
 
-    pub fn set_project(config: &OckamConfig, project: &Project) -> Result<()> {
+    async fn set(config: &OckamConfig, project: &Project<'_>) -> Result<()> {
         if !project.is_ready() {
             return Err(anyhow!(
                 "Project is not ready yet, wait a few seconds and try again"
             ));
         }
+        let proute: MultiAddr = project
+            .access_route
+            .as_ref()
+            .try_into()
+            .context("Invalid project node route")?;
+        let pid = project
+            .identity
+            .as_ref()
+            .context("Project should have identity set")?;
+        let authority = if let Some(r) = &project.authority_access_route {
+            let rte = MultiAddr::try_from(&**r).context("Invalid project authority address")?;
+            let a = project
+                .authority_identity
+                .as_ref()
+                .context("Missing project authority")?;
+            let a = hex::decode(&**a).context("Invalid project authority")?;
+            let v = Vault::default();
```

Comment:
I don't like that we need to create a Vault here, but let's not bother with that problem for now